Testing

PAS195Pitt opened the pull request:
first few files with jsdoc3 documentation

PAS195Pitt committed:
testing

PAS195Pitt committed:
Added JSDoc3 documentation to some files

PAS195Pitt committed:
jsdoc3 added

pearcetm commented on js/addcss.mjs:
```diff
@@ -1,3 +1,11 @@
+ *    the CSS file will only be added if the pattern is found in the URL.
+ * @returns {void}
+ */
 function addCSS(url, nameToCheck){
```
This is a good simple documentation of the function.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -1,9 +1,16 @@
+     * @param {any} atk - The ATK object.
+     * @param {object} opts - Additional options for the file dialog.
+     */
+    constructor(atk, opts){        
```
There is no such thing as "the ATK object". In the code, atk is short for annotation toolkit. We don't want to confuse things in the documentation.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -34,13 +55,36 @@ export class FileDialog{
+             *
+             * @type {jQuery}
+             */
             let finput = $('<input>',{type:'file',accept:'text/geojson,.geojson,text/json,.json'});
```
Comments that explain code are fine but they do not need to be in JSDoc format like this. Only function definitions, and really only public API functions, should be documented this way.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -34,13 +55,36 @@ export class FileDialog{
+                 * The parsed GeoJSON data.
+                 *
+                 * @type {Array|Object}
+                 */    
```
See previous comment - this is not appropriate documentation for within a function.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -80,52 +124,67 @@ export class FileDialog{
+         * @function
+         * @memberof AtkMap
+         * @param {Object} atk - The AtkMap object that contains the features.
+         * @returns {void}
```
It's fine to document functions like this

pearcetm commented on js/filedialog.mjs:
```diff
@@ -80,52 +124,67 @@ export class FileDialog{
+            let filename=getFileName(); // Get the file name from the AtkMap object
+            let lskeys=Object.keys(window.localStorage); // Get the keys from the local storage
+            let list = _this.element.find('.featurecollection-list').empty(); // Find and empty the feature collection list element
+            let div=$('<div>',{class:'localstorage-key-list'}).appendTo(list); // Create a div element for the local storage key list and append it to the list element
```
But it's overkill to comment on every line like is done here and below.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -80,52 +124,67 @@ export class FileDialog{
+         * Loads the feature collections from the local storage.
+         * @function
+         * @memberof AtkMap
+         * @param {Object} atk - The AtkMap object that contains the features.
```
There is no such construct as an `AtkMap` in the code, this needs to be changed (here and elsewhere)

pearcetm commented on js/filedialog.mjs:
```diff
@@ -147,40 +206,73 @@ export class FileDialog{
+         * @memberof AtkMap
+         */
+        function exportSVG(){
+            initDlg(); // Initialize the dialog box
```
`exportPNG` has been totally deleted and `exportSVG` has been duplicated. This needs to be fixed.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -147,40 +206,73 @@ export class FileDialog{
+                    toSave.forEach(function(s){ // For each feature collection
+                        p.project.addLayer(s.layer.clone({insert:false,deep:true})); // Add a cloned layer to the paper project
                     })
-                    // let blob = new Blob([p.project.activeLayer.rasterize({insert:false}).toDataURL()],{type:'image/png'});
```
Don't delete comments like this at this point. They have use in hinting at how the code could be used.

pearcetm commented on js/filedialog.mjs:
```diff
@@ -147,40 +206,73 @@ export class FileDialog{
         }
-        function localstorageStore(){
-            initDlg();
-            let fcs = atk.toGeoJSON();
```
Why was `localstorageStore()` deleted?

pearcetm commented on js/rotationcontrol.mjs:
```diff
@@ -1,103 +1,190 @@
+        /**
+         * Handles the mouse down event.
+         * @param {Object} ev - The event object.
+         */
```
Not incorrect, but not part of the public API and does not need this style of documentation

pearcetm commented on js/rotationcontrol.mjs:
```diff
@@ -1,103 +1,190 @@
 }
 
+/**
+ * Creates a widget for controlling the rotation of the map.
```
What map?

pearcetm commented on js/rotationcontrol.mjs:
```diff
@@ -180,7 +267,7 @@ function RotationControlWidget(center, setAngle){
     group.addChild(rotationLineControl);
     group.pivot = circle.bounds.center;//make the center of the circle the pivot for the entire  controller
-    group.position = center;//set position after adding all children so it is applied to all
+    group.position = center;
```
This was a useful comment explaining why the line of code had to be after a previous line, and shouldn't have been deleted.

pearcetm reviewed (changes requested):
Not a bad start! But definitely some things that need a human touch here. I've added some comments at specific places, but many of these need to be applied throughout the documents, not only where I've highlighted.

PAS195Pitt commented:
Hello,

I did notice that chatgpt was struggling a lot with documenting the programs, especially since the programs are split into different files. I noticed it would also start removing code. So instead, I started filling some of them manually that chatgpt struggled with. I will fix up the files I sent and send you updated versions this weekend. The comments you sent were also very helpful and I'll apply the fixes you mentioned in them.

Thank you,
Paul Sharoubeem

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: pearcetm ***@***.***>
Sent: Friday, June 30, 2023 4:00:34 PM
To: pearcetm/osd-paperjs-annotation ***@***.***>
Cc: Sharoubeem, Paula Bassem ***@***.***>; Author ***@***.***>
Subject: Re: [pearcetm/osd-paperjs-annotation] Testing (PR #1)


@pearcetm requested changes on this pull request.

Not a bad start! But definitely some things that need a human touch here. I've added some comments at specific places, but many of these need to be applied throughout the documents, not only where I've highlighted.

________________________________

In js/addcss.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248228259>:

> +/**
+ * Adds a CSS file to the document head if it is not already present.
+ *
+ * @param {string} url - The URL of the CSS file to add.
+ * @param {string} [nameToCheck] - The name pattern to check in the URL. If provided,
+ *    the CSS file will only be added if the pattern is found in the URL.
+ * @returns {void}
+ */
 function addCSS(url, nameToCheck){


This is a good simple documentation of the function.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248228846>:

> +    /**
+     * Creates an instance of the FileDialog.
+     *
+     * @param {any} atk - The ATK object.
+     * @param {object} opts - Additional options for the file dialog.
+     */
+    constructor(atk, opts){


There is no such thing as "the ATK object". In the code, atk is short for annotation toolkit. We don't want to confuse things in the documentation.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248229863>:

> +            /**
+             * Represents a file input element for selecting GeoJSON files.
+             *
+             * @type {jQuery}
+             */
             let finput = $('<input>',{type:'file',accept:'text/geojson,.geojson,text/json,.json'});


Comments that explain code are fine but they do not need to be in JSDoc format like this. Only function definitions, and really only public API functions, should be documented this way.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248230240>:

> +                /**
+                 * The selected file.
+                 *
+                 * @type {File}
+                 */
+            // console.log('File picked',this.files[0]);
                 let file = this.files[0];
+                /**
+                 * Represents a FileReader object for reading file data.
+                 *
+                 * @type {FileReader}
+                 */
                 let fr = new FileReader();
+                /**
+                 * The parsed GeoJSON data.
+                 *
+                 * @type {Array|Object}
+                 */


See previous comment - this is not appropriate documentation for within a function.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248230966>:

> +
+        /**
+         * Loads the feature collections from the local storage.
+         * @function
+         * @memberof AtkMap
+         * @param {Object} atk - The AtkMap object that contains the features.
+         * @returns {void}


It's fine to document functions like this

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248231245>:

> +         */
+        function localstorageLoad(atk){
+            initDlg(); // Initialize the dialog box
+            let geoJSON=[]; // An array to store the GeoJSON data
+            let filename=getFileName(); // Get the file name from the AtkMap object
+            let lskeys=Object.keys(window.localStorage); // Get the keys from the local storage
+            let list = _this.element.find('.featurecollection-list').empty(); // Find and empty the feature collection list element
+            let div=$('<div>',{class:'localstorage-key-list'}).appendTo(list); // Create a div element for the local storage key list and append it to the list element


But it's overkill to comment on every line like is done here and below.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248232016>:

> +        /**
+         * Loads the feature collections from the local storage.
+         * @function
+         * @memberof AtkMap
+         * @param {Object} atk - The AtkMap object that contains the features.


There is no such construct as an AtkMap in the code, this needs to be changed (here and elsewhere)

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248232783>:

>          }
-        function exportPNG(){
-            initDlg();
-            let fcs = atk.getFeatureCollectionLayers();
-            let list = setupFeatureCollectionList(fcs);
-            let finishbutton = setupFinalize('Create file','Choose file name:',getFileName()+'-raster.png');
-            finishbutton.on('click',function(){
-                $(this).parent().find('.download-link').remove();
-                let toSave=list.find('input:checked').toArray().map(function(cb){return $(cb).data('fc')});
-                if(toSave.length>0){
-                    let p = new paper.PaperScope();
-                    p.setup();
-                    toSave.forEach(function(s){
-                        p.project.activeLayer.addChildren(s.layer.clone({insert:false,deep:true}).children);
+        /**
+         * Exports the feature collections as a SVG file.
+         * @function
+         * @memberof AtkMap
+         */
+        function exportSVG(){
+            initDlg(); // Initialize the dialog box


exportPNG has been totally deleted and exportSVG has been duplicated. This needs to be fixed.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248233376>:

> +                        p.project.addLayer(s.layer.clone({insert:false,deep:true})); // Add a cloned layer to the paper project
                     })
-                    // let blob = new Blob([p.project.activeLayer.rasterize({insert:false}).toDataURL()],{type:'image/png'});


Don't delete comments like this at this point. They have use in hinting at how the code could be used.

________________________________

In js/filedialog.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248234119>:

>              })
         }
-        function localstorageStore(){
-            initDlg();
-            let fcs = atk.toGeoJSON();


Why was localstorageStore() deleted?

________________________________

In js/rotationcontrol.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248240201>:

> +        /**
+         * Handles the mouse down event.
+         * @param {Object} ev - The event object.
+         */


Not incorrect, but not part of the public API and does not need this style of documentation

________________________________

In js/rotationcontrol.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248240903>:

> +/**
+ * Creates a widget for controlling the rotation of the map.


What map?

________________________________

In js/rotationcontrol.mjs<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#discussion_r1248241357>:

>      group.pivot = circle.bounds.center;//make the center of the circle the pivot for the entire  controller
-    group.position = center;//set position after adding all children so it is applied to all
+    group.position = center;


This was a useful comment explaining why the line of code had to be after a previous line, and shouldn't have been deleted.

—
Reply to this email directly, view it on GitHub<https://github.com/pearcetm/osd-paperjs-annotation/pull/1#pullrequestreview-1507798923>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVKTNYJNLAZTES6J3ZEYHXDXN4V6FANCNFSM6AAAAAAZXYKA5E>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

PAS195Pitt committed:
Updated JSdoc3

PAS195Pitt committed:
Updated JSdoc3

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -14,18 +14,24 @@ import {TransformTool} from './papertools/transform.mjs';
 import {RasterTool} from './papertools/raster.mjs';
 
 export class AnnotationToolbar{
```
It would be great to document what the class itself actually does with a class description. See the [OpenSeadragon](https://github.com/openseadragon/openseadragon) source code for an example (like here: https://github.com/openseadragon/openseadragon/blob/master/src/navigator.js). Note that those "classes" were implemented before classes were a real javascript construct so it might work a little differently - but probably not much.

pearcetm commented on js/addcss.mjs:
```diff
@@ -1,3 +1,11 @@
+ * Adds a CSS file to the document head if it is not already present.
+ *
+ * @param {string} url - The URL of the CSS file to add.
+ * @param {string} [nameToCheck] - The name pattern to check in the URL. If provided,
+ *    the CSS file will only be added if the pattern is found in the URL.
```
This is only part of the logic of the `nameToCheck` parameter. It should read "A regular expression pattern to check for in existing CSS links in the document, used to avoid duplicating CSS files. If the pattern is found in the document, the new link is not added. For consistency, the pattern is also checked against the url parameter to make sure it will exist in the document after adding the new link. If the pattern is not found in the url, an error is logged to the console and the file is not added."

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -98,41 +106,58 @@ export class AnnotationToolbar{
     
-    setMode(){
-        let self=this;
+    /**
+     * Sets the current mode of the Annotation Toolbar based on the selection and active tool.
```
This comment is not accurate. This should read "Sets the mode of the toolbar based on the currently selected items in the project. Individual tools will be enabled and disabled by this. If the currently active tool is not supported for the selected item(s) it will be deactivated." (The active tool is not used in making the decision of which tools should be enabled or disabled.)

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -98,41 +106,58 @@ export class AnnotationToolbar{
+     * @memberof AnnotationToolbar
+     * @instance
+     * @function setMode
+     * @throws {Error} Throws an error if the active tool's toolbar control is not found.
```
This is not accurate - no error is thrown.

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -98,41 +106,58 @@ export class AnnotationToolbar{
-            })
-            activeTool.selectionChanged();
+        this.setModeTimeout = setTimeout(() => {
+        this.setModeTimeout = null;
+        let selection = this.paperScope.findSelectedItems();
+        let activeTool = this.paperScope.getActiveTool();
+        if (selection.length === 0) {
+            this.currentMode = 'select';
+        } else if (selection.length === 1) {
+            let item = selection[0];
+            let def = item.annotationItem || {};
+            let type = def.type;
+            if (def.subtype) type += ':' + def.subtype;
+            let mode = type === null ? 'new' : type;
+            this.currentMode = mode;
+        } else {
+            this.currentMode = 'multiselection';
+        }
+    
+        if (activeTool.getToolbarControl().isEnabledForMode(this.currentMode) === false) {
+            activeTool.deactivate(true);
+            this.tools.default.activate();
+        }
+    
+
+        Object.values(this.tools).forEach(toolObj => {
+            let t = toolObj.getToolbarControl();
+            t && (t.isEnabledForMode(self.currentMode) ? t.button.enable() : t.button.disable());
+        });
+    
+
+        activeTool.selectionChanged();
```
This code should not have been un-indented.

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -144,6 +169,15 @@ export class AnnotationToolbar{
     hide(){
         $(this.ui.buttongroup.element).hide();
     }
```
This function should be documented.

pearcetm replied:
This function should be documented.

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -98,41 +106,58 @@ export class AnnotationToolbar{
+ * @instance
+ * @function addToolbarControl
+ * @param {Object} toolbarControl - The toolbar control to be added.
+ * @throws {Error} Throws an error if the toolbar control's button element is not found.
```
This is incorrect - no error is thrown.

pearcetm commented on js/annotationtoolbar.mjs:
```diff
@@ -144,6 +169,15 @@ export class AnnotationToolbar{
         $(this.ui.buttongroup.element).hide();
     }
+    
+/**
+ * Adds the Annotation Toolbar to an OpenSeadragon viewer.
+ *
+ * @memberof AnnotationToolbar
+ * @instance
+ * @function addToOpenSeadragon
+ * @param {Object} viewer - The OpenSeadragon viewer.
+ */
```
Additional information about how and where the toolbar is added would be useful (i.e. an OpenSeadragon.ButtonGroup is created and added to the OpenSeadragon.ControlAnchor.TOP_LEFT position). Also mention that the .viewer-controls-topleft class is added.

PAS195Pitt committed:
Gulpfile integrated with webpack, jsdoc, and other functions
File directory changed to src/js
Updated jsdocumentation, need to review PaperOverlay, RotationControl, and discuss proper documentation

PAS195Pitt committed:
updated readme

PAS195Pitt committed:
Reconfigured gulp jsdoc task so that we arent using gulp-jsdoc3 but actual jsdco3 and jsdoc.conf.json instead

Added sources to documentation and line references

Fixed static misclassification

PAS195Pitt committed:
Created Demos for every tool, still need to add example usage with brain tissues

Documented all papertools, might need extra tweaking

PAS195Pitt committed:
testing new documentation styles

PAS195Pitt committed:
checking how module tag works

PAS195Pitt committed:
testing module

PAS195Pitt committed:
Added Documentation to all Paper Items
Fixed private documentations and added missing documentation to Paper Tools

PAS195Pitt committed:
updated demos with simple guided examples
